runtime/runtime.go

```diff
@@ -104,6 +108,9 @@ func disable(name string) error {
 		return ErrNoExist
 	}
 
+	fp.failpointMu.Lock()
```

Comment:
Checked the codebase and found that `disable` is called with `failpointsMu`. It seems we don't need to add failpoint-level mutex?

Reply:
Same with `Status`. My idea is just to make sure we do the fine-grain FailPoint locking properly, so in the case where we change the locking mechanism on the `failpointsMu` level, we are guaranteed that the `failpointMu` lock will still function as we expected. 

Does this make sense to you? :)

Reply:
Maybe another change we can do is to really lock `failpointsMu` only when the map is being accessed, but after accessing the map we release `failpointsMu`.